sys/kern/kern_tc.c

```diff
@@ -1769,15 +1770,16 @@ pps_capture(struct pps_state *pps)
 #ifdef FFCLOCK
 	pps->capffth = fftimehands;
 #endif
-	pps->capcount = th->th_counter->tc_get_timecount(th->th_counter);
-	atomic_thread_fence_acq();
-	if (pps->capgen != th->th_generation)
-		pps->capgen = 0;
```

Comment:
Here we can remove the check because we have the same condition in the check in line 1811 (after changes).